schemas/filetype.yaml

```
@@ -48,6 +48,13 @@ classes:
                 description: >-
                     A list of any known software (proprietary or otherwise) that produces
                     such `FileType`.
+            associated_file_extensions:
+                multivalued: true
+                description: >-
+                    A list of any known file extensions that this file is found with,
+                    to be used
+                    as a hint for file type detection. Should omit the leading '.',
+                    e.g., 'json' or 'txt'.
```

Ignoring formatting, I'd suggest something like:
```suggestion
                    A list of any known file extensions that files of this `FileType` are found with. These may be used as a hint for `FileType` detection. Should omit the leading `'.'`, e.g. 'json' or 'txt'.
```

---

I had one remaining question about whether we should expand this to something like filename patterns (following Markus' work), to be able to do things like  `POSCAR` which would match the entire filename and `*.json` would match other JSON files. Would need to be specifically using Python's regex engine though.

---

I think since the key is `associated_file_extensions`, we can always safely assume it's an extension (i.e. period-separated). Maybe later we can have `associated_file_names` for something like `POSCAR`, but whether that's useful for anything else is a good question.